### ISSUE TITLE
Missing digital signatures on latest updates downloaded via Docker Desktop update mechanism

### ISSUE THREAD

**Issue by wondering639** [docker/for-win#14396]
### Description

On later versions, I think starting around 4.35.0, updates downloaded directly via Docker Desktop are missing a digital signature. This is not the case when downloading the updates/installer directly from https://docs.docker.com/desktop/release-notes/

### Reproduce

Download update via Docker Desktop -> Settings -> Software updates, apply&restart. Windows will show a UAC hint about an unknown publisher.

### Expected behavior

No hint about an unknown publisher, instead Docker should be shown as publisher

### docker version

Only Docker Desktop relevant, 4.35.0 and later

### docker info

Only Docker Desktop relevant, 4.35.0 and later

### Diagnostics ID

not relevant

**Comment by f2bo** [comment]
I have the same issue. It seems to be a recurring problem.

https://github.com/docker/for-win/issues/14381
https://github.com/docker/for-win/issues/13574
https://github.com/docker/for-win/issues/13594

**Comment by ktzsolt** [comment]
Can confirm.

Workaround is to download the full installer from here: https://docs.docker.com/desktop/release-notes/

I installed with this on top of an existing DD installation (turned off/exited DD first) and asked if I want to replace my old version with the new version, clicked yes and the install basically did an upgrade, docker-desktop-data was not replaced so all the data, images, build cache, volumes etc are kept.

**Comment by maffe** [comment]
I deleted the Docker…exe found in %tmp%\DockerDesktopUpdates\, after that Docker Desktop downloaded the update again and the UAC prompt showed Docker as the publisher (update: this method initially did not work with the update from 4.35.1 to 4.36.0, but after a few days it did).

Previously it was [suggested](https://github.com/docker/for-win/issues/14381#issuecomment-2441003973) to run `Get-bitstransfer –allusers | remove-bitstransfer` (in elevated PowerShell) but `Get-BitsTransfer -allusers` did not show anything Docker related.

**Comment by wondering639** [comment]
seems to be fixed on later updates, at least for me. can others confirm?

**Comment by Bogay** [comment]
The downloaded update is still unsigned on my side. Tested version: 4.36.0 (175267)

**Comment by f2bo** [comment]
Nothing has changed in my case as well. I still see an unsigned update when attempting to upgrade from 4.34.3 to 4.36.0.

**Comment by maffe** [comment]
@Bogay @f2bo Did you delete the unsigned file from %tmp%\DockerDesktopUpdates? The newer, signed files will not be downloaded if there is an unsigned file from a previous update attempt.

**Comment by f2bo** [comment]
@maffe Yes. I did, and several times since. I did not run the full installer though, as suggested above. 

![Image](https://github.com/user-attachments/assets/8efe1061-3861-42ae-a3fe-77eca91cf43f)

**Comment by THBrew** [comment]
We're seeing the same issue as of today; December 12th, 2024.

**Comment by maffe** [comment]
I had this problem twice (with an update to 4.36.0 and an earlier version) and had to delete the unsigned installers and wait for signed ones, but today’s update to 4.37.0 came directly with a signed installer.

**Comment by ktzsolt** [comment]
> I had this problem twice (with an update to 4.36.0 and an earlier version) and had to delete the unsigned installers and wait for signed ones, but today’s update to 4.37.0 came directly with a signed installer.

Same for me, it looks good now.

**Comment by f2bo** [comment]
The update process can clearly be improved.

Initially, while I could see the new 4.37.0 release [here](https://docs.docker.com/desktop/release-notes/#4370), I was still offered 4.36.0 in Docker Desktop and if I tried to install that, again, it was still unsigned. Before anyone asks, I restarted Docker and tried a second time with the same result.

![Image](https://github.com/user-attachments/assets/10248aa4-1291-411c-b8af-86b89ea2a749)

Next, I disabled **Automatically check for updates** and restarted Docker Desktop. Once it was running again, it told me that 4.34.3 was the newest version available and that I was up to date. 

![Image](https://github.com/user-attachments/assets/4232f07e-2e49-4002-ac73-2004121214de)

Then, after I clicked **Check for updates**, it finally started showing 4.37.0.

![Image](https://github.com/user-attachments/assets/c6402258-caee-4980-8d64-2699fbcbf5d7)

Thankfully, this update was signed and I installed it successfully.

**Comment by Bogay** [comment]
> Did you delete the unsigned file from %tmp%\DockerDesktopUpdates? The newer, signed files will not be downloaded if there is an unsigned file from a previous update attempt.

Hi @maffe,
Deleting the unsigned file from `%tmp%\DockerDesktopUpdates` worked for me. The UI shows a progress bar indicating that Docker Desktop is downloading the release file, but it seems that it isn't actually doing so. Now I have successfully upgraded to 4.37.0, thanks.
